### PR TITLE
Check if cluster is set before checking VDS membership

### LINKED ISSUE #1860: vic-machine return nil error
If there are multiple VC clusters exists, and user didn't specify a correct compute-resource, vic-machine create return error message like following:

invalid memory address or nil pointer dereference

### PR DESCRIPTION
**Pull request by andrewtchin**
Fixes #1860 

Before:
```
⇒  bin/vic-machine-linux create --target 10.17.109.115 --image-datastore vsanDatastore --name chin-vic-18 --user administrator@vsphere.local --password password --compute-resource chin --bridge-network vch-test --timeout 10m                                                                                                                                  
INFO[2016-08-10T14:06:29-07:00] ### Installing VCH ####                      
INFO[2016-08-10T14:06:29-07:00] Generating certificate/key pair - private key in ./chin-vic-18-key.pem 
INFO[2016-08-10T14:06:30-07:00] Validating supplied configuration            
INFO[2016-08-10T14:06:30-07:00] Suggesting valid values for --compute-resource based on "/dc1/host/chin/Resources" 
INFO[2016-08-10T14:06:30-07:00] Suggested values for --compute-resource:     
INFO[2016-08-10T14:06:30-07:00]   "cluster1"                                 
INFO[2016-08-10T14:06:30-07:00]   "cluster2"                                 
ERRO[2016-08-10T14:06:31-07:00] --------------------                         
ERRO[2016-08-10T14:06:31-07:00] vic-machine-linux failed: runtime error: invalid memory address or nil pointer dereference
```

Now:
```
⇒  bin/vic-machine-linux create --target 10.17.109.115 --image-datastore vsanDatastore --name chin-vic-18 --user administrator@vsphere.local --password password --compute-resource chin --bridge-network vch-test --timeout 10m
INFO[2016-08-11T07:06:15-07:00] ### Installing VCH ####                      
INFO[2016-08-11T07:06:15-07:00] Generating certificate/key pair - private key in ./chin-vic-18-key.pem 
INFO[2016-08-11T07:06:16-07:00] Validating supplied configuration            
INFO[2016-08-11T07:06:16-07:00] Suggesting valid values for --compute-resource based on "/dc1/host/chin/Resources" 
INFO[2016-08-11T07:06:17-07:00] Suggested values for --compute-resource:     
INFO[2016-08-11T07:06:17-07:00]   "cluster1"                                 
INFO[2016-08-11T07:06:17-07:00]   "cluster2"                                 
ERRO[2016-08-11T07:06:17-07:00] Firewall check SKIPPED                       
ERRO[2016-08-11T07:06:17-07:00]   cluster not set                            
ERRO[2016-08-11T07:06:17-07:00] License check SKIPPED                        
ERRO[2016-08-11T07:06:17-07:00]   cluster not set                            
ERRO[2016-08-11T07:06:17-07:00] DRS check SKIPPED                            
ERRO[2016-08-11T07:06:17-07:00]   cluster not set                            
ERRO[2016-08-11T07:06:17-07:00] Compatibility check SKIPPED                  
ERRO[2016-08-11T07:06:17-07:00]   cluster not set                            
ERRO[2016-08-11T07:06:17-07:00] --------------------                         
ERRO[2016-08-11T07:06:17-07:00] resource pool '/dc1/host/chin/Resources' not found 
ERRO[2016-08-11T07:06:17-07:00] Unable to check hosts in vDS for "vch-test": cluster not set 
ERRO[2016-08-11T07:06:17-07:00] Firewall check SKIPPED                       
ERRO[2016-08-11T07:06:17-07:00] License check SKIPPED                        
ERRO[2016-08-11T07:06:17-07:00] DRS check SKIPPED                            
ERRO[2016-08-11T07:06:17-07:00] Compatibility check SKIPPED                  
ERRO[2016-08-11T07:06:17-07:00] Create cannot continue: configuration validation failed 
ERRO[2016-08-11T07:06:17-07:00] --------------------                         
ERRO[2016-08-11T07:06:17-07:00] vic-machine-linux failed: validation of configuration failed
```

### CODE REVIEW

File: lib/install/validate/network.go [hunk 1]
```diff
@@ -317,6 +317,9 @@ func (v *Validator) checkVDSMembership(ctx context.Context, network types.Manage
 		return nil
 	}
 
+	if v.Session.Cluster == nil {
+		return errors.New("cluster not set")
```
Comment (emlin) [inline lib/install/validate/network.go]: cluster is not a value set by user, might have more message, suggest user to check compute-resource settings.

**Comment by emlin** [comment]
lgtm

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**